Sources/SQLite/SQLiteError.swift

```diff
@@ -12,33 +16,100 @@ public enum SQLiteError: Error, Equatable {
     case onWrite(Array<SQLiteRow>)
     case onGetColumnType(Int32)
     case onBeginTransactionAfterDeallocating
-    case onExecuteQueryWithoutOpenDatabase
     case onExecuteQueryAfterDeallocating
-    case onCreateFunction(String, Int32)
-    case onRemoveFunction(String, Int32)
     case onGetColumnInTable(String)
     case onGetIndexInTable(String)
     case onGetSQL
     case onInvalidTableName(String)
     case onDecodingRow(String)
     case onInvalidDecodingType(String)
     case onInvalidSelectStatementColumnCount
-    case onObserveWithoutColumnMetadata
+    case onSubscribeWithoutColumnMetadata
     case onSubscribeWithoutDatabase
     case onTryToObserveZeroTables
 }
 
+extension SQLiteError {
+    public var code: Int32? {
+        switch self {
+        case .databaseIsClosed:
+            return nil
+        case .onInternalError:
+            return nil
+        case .onInvalidPath:
+            return nil
+        case .onOpenSharedDatabase:
+            return nil
+        case let .onOpen(code, _):
+            return code
+        case let .onEnableWAL(code):
+            return code
+        case let .onClose(code):
+            return code
+        case let .onPrepareStatement(code, _):
+            return code
+        case .onGetParameterIndex:
+            return nil
+        case let .onBindParameter(code, _, _):
+            return code
+        case let .onStep(code, _):
+            return code
+        case .onWrite:
+            return nil
+        case .onGetColumnType:
+            return nil
+        case .onBeginTransactionAfterDeallocating:
+            return nil
+        case .onExecuteQueryAfterDeallocating:
+            return nil
+        case .onGetColumnInTable:
+            return nil
+        case .onGetIndexInTable:
+            return nil
+        case .onGetSQL:
+            return nil
+        case .onInvalidTableName:
+            return nil
+        case .onDecodingRow:
+            return nil
+        case .onInvalidDecodingType:
+            return nil
+        case .onInvalidSelectStatementColumnCount:
+            return nil
+        case .onSubscribeWithoutColumnMetadata:
+            return nil
+        case .onSubscribeWithoutDatabase:
+            return nil
+        case .onTryToObserveZeroTables:
+            return nil
+        }
+    }
+
+    public var isBusy: Bool {
```

Comment:
This isn't used yet, but it will be if we need to add special retry handling for busy errors.